clients/drcachesim/reader/v2p_reader.h

```diff
@@ -32,27 +32,41 @@
 
 /* v2p_reader: reads and parses a virtual-to-physical address mapping in textproto format.
  * Creates virtual-to-physical address map in memory.
+ * The section of the textproto file that we are interested in parsing is a sequence of
+ * blocks that follow this format:
+ * address_mapping {
+    virtual_address: 0x123
+    physical_address: 0x3
+ * }
+ * In gen_v2p_map() we rely on the fact that virtual_address and physical_address are one
+ * after the other on two different lines.
  */
 
 #ifndef _V2P_READER_H_
 #define _V2P_READER_H_ 1
 
-#include <stdint.h>
+#include "trace_entry.h"
 
 #include <cstdint>
-#include <unordered_map>
 #include <string>
+#include <unordered_map>
 
 namespace dynamorio {
 namespace drmemtrace {
 
+struct v2p_info_t {
+    uint64_t page_count;
+    uint64_t bytes_mapped;
+    size_t page_size;
+    std::unordered_map<addr_t, addr_t> v2p_map;
+};
+
 class v2p_reader_t {
 public:
-    v2p_reader_t();
+    v2p_reader_t() = default;
 
-    bool
-    gen_v2p_map(std::string path_to_file,
-                std::unordered_map<uint64_t, uint64_t> &v2p_map);
+    std::string
+    init_v2p_info_from_file(std::string path_to_file, v2p_info_t &v2p_info);
```

Comment:
optional?: s/init/read/ or s/init/create/?  Somehow "init" doesn't sound like the whole process: it sounds like it would just zero initial fields or sthg.

Reply:
Going with "create_". Done.